### PR TITLE
Update TypeScript API Route example

### PR DESCRIPTION
**Pull request by abir-taheer**
The existing example causes issues with eslint's `eslintimport/no-anonymous-default-export` rule that come pre-enabled with `create-next-app`.

Added a name to the api handler function before setting it as the default export.

## Documentation / Examples

- [x] Make sure the linting passes by running `pnpm build && pnpm lint`
- [x] The "examples guidelines" are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing/examples/adding-examples.md)

### CODE REVIEW

File: docs/basic-features/typescript.md [hunk 1]
```diff
@@ -103,9 +103,11 @@ The following is an example of how to use the built-in types for API routes:
 ```ts
 import type { NextApiRequest, NextApiResponse } from 'next'
 
-export default (req: NextApiRequest, res: NextApiResponse) => {
+const handler = (req: NextApiRequest, res: NextApiResponse) => {
   res.status(200).json({ name: 'John Doe' })
 }
+
+export default handler;
```
Comment (timneutkens) [inline docs/basic-features/typescript.md:110]:
```suggestion
export default function handler(req: NextApiRequest, res: NextApiResponse) {
  res.status(200).json({ name: 'John Doe' })
}
```

File: docs/basic-features/typescript.md [hunk 2]
```diff
@@ -117,9 +119,11 @@ type Data = {
   name: string
 }
 
-export default (req: NextApiRequest, res: NextApiResponse<Data>) => {
+const handler = (req: NextApiRequest, res: NextApiResponse<Data>) => {
   res.status(200).json({ name: 'John Doe' })
 }
+
+export default handler;
```
Comment (timneutkens) [inline docs/basic-features/typescript.md:126]:
```suggestion
export default function handler(req: NextApiRequest, res: NextApiResponse<Data>) {
  res.status(200).json({ name: 'John Doe' })
}
```

**Review by timneutkens — DISMISSED** [review]
Left two changes to use `function` instead of `const` to be in line with other examples for e.g. React components.

**Review by ijjk — APPROVED** [review]
Thanks!

**Pull request merged**